.github/actions/go-e2e/action.yaml

```diff
@@ -0,0 +1,50 @@
+name: 'Go E2E Tests Reusable Action'
+description: 'Run Go E2E tests using the provided TAS server and Go version.'
+inputs:
+  tas_address:
+    description: 'TAS server address'
+    required: true
+    type: string
+  go_version:
+    description: 'Go version'
+    required: true
+    default: '1.21'
+    type: string
+runs:
+  using: 'composite'
+  steps:
+    - name: Checkout go test source code
+      uses: actions/checkout@v4
+      with:
+        repository: "securesign/sigstore-e2e"
+        path: e2e
+    - name: Install Go
+      uses: actions/setup-go@v5
+      with:
+        go-version: ${{ inputs.go_version }}
+
+    - name: Resolve TAS_SINGLENODE hostname
+      shell: bash
+      run: |
+        echo "${{ inputs.tas_address }} fulcio.myrhtas tuf.myrhtas rekor.myrhtas tsa.myrhtas rekor-search.myrhtas cli-server.myrhtas dex-idp" | sudo tee -a /etc/hosts
```

Comment:
Nit: for this step and for the last one, could we have variables defining the service hostnames? We're hardcoding a bunch of values in two places.